perfkitbenchmarker/disk.py

```diff
@@ -18,14 +18,53 @@
 """
 
 import abc
+import logging
 
+from perfkitbenchmarker import flags
 from perfkitbenchmarker import resource
 
+FLAGS = flags.FLAGS
+
+
+# These are the (deprecated) old disk type names
 STANDARD = 'standard'
 REMOTE_SSD = 'remote_ssd'
 PIOPS = 'piops'  # Provisioned IOPS (SSD) in AWS
+
+# 'local' refers to disks that come attached to VMs. It is the only
+# "universal" disk type that is not associated with a provider. It
+# exists because we can provision a local disk without creating a disk
+# spec. The Aerospike benchmarks use this fact in their config
+# methods, and they need to be able to tell when a disk is local. So
+# until that changes, 'local' is a special disk type.
 LOCAL = 'local'
 
+# Map old disk type names to new disk type names
+DISK_TYPE_MAPS = dict()
+
+# Standard metadata keys relating to disks
+MEDIA = 'media'
+REPLICATION = 'replication'
+# And some possible values
+HDD = 'hdd'
+SSD = 'ssd'
+NONE = 'none'
+ZONE = 'zone'
+REGION = 'region'
+
+
+def RegisterDiskTypeMap(provider_name, map):
+  """Register a map from legacy disk type names to modern ones.
+
+  The translation machinery looks here to find the map corresponding
+  to the chosen provider and translates the user's flags and configs
+  to the new naming system. This function should be removed once the
+  (deprecated) legacy flags are removed.
+  """
+
+  DISK_TYPE_MAPS[provider_name] = map
```

Comment:
Use of the parameter name `map` shadows the `map()` function in Python. Please change parameter name.